core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -63,96 +70,106 @@ angular.module('oppia').factory('EditableStoryBackendApiService', [
         commit_message: commitMessage,
         change_dicts: changeList
       };
-      $http.put(editableStoryDataUrl, putData).then(function(response) {
+
+      this.http.post(
+        editableStoryDataUrl, putData).toPromise().then(
+        (response: any) => {
         // The returned data is an updated story dict.
-        var story = angular.copy(response.data.story);
+          var story = angular.copy(response.body.story);
```

Comment:
Use cloneDeep (see point 6 [here](https://github.com/oppia/oppia/wiki/Angular-Migration#implementation-details-to-migrate-services)).

Also, drop "body" and make it `response.story`.

Reply:
Done.